sapio-base/src/plugin_args.rs

```diff
@@ -32,6 +33,11 @@ pub struct CreateArgs<S> {
     /// Others arguments set by general system settings
     pub context: ContextualArguments,
 }
+impl<S: HasSapioModuleSchema> HasSapioModuleSchema for CreateArgs<S> {
```

Comment:
The trait HasSapioModuleSchema being impl for both `CreateArgs<S>` and S seems funky. Probably should be something where if we want the trait SapioModuleSchema it should only be defined over CreateArgs<S> if that is indeed the thing we are calling "MoudleSchema"?